src/build-esm.ts

```diff
@@ -13,34 +13,57 @@ import tsc from './which-tsc.js'
 const node = process.execPath
 const { esmDialects = [] } = config
 
-export const buildESM = () => {
+const isBuildResult = (x: any): x is BuildResult => {
```

Comment:
Style nit, I usually prefer to define `isType` methods in the same file as the type itself, to reduce the chances of having them go out of sync. (No need to change now, I'm fine moving it later, so this is more a note to myself.)